synapse/tools/axon/load.py

```diff
@@ -0,0 +1,117 @@
+import synapse.lib.output as s_output
+import synapse.lib.msgpack as s_msgpack
+
+descr = '''
+Load blobs into a Synapse Axon.
+'''
+
+async def loadBlobs(opts, outp, blobsfiles):
+    try:
+        async with await s_telepath.openurl(opts.url) as axon:
+
+            cellinfo = await axon.getCellInfo()
+            celltype = cellinfo['cell']['type']
+            if "axon" not in celltype.lower():
+                mesg = f'Axon load tool only works on axons, not {celltype}'
+                raise s_exc.TypeMismatch(mesg=mesg)
+
+            for blobsfile in blobsfiles:
+                outp.printf(f'Processing blobs file: {blobsfile}')
+                with open(blobsfile, 'rb') as fd:
+                    msgit = s_msgpack.iterfd(fd)
+                    for mesg in msgit:
+                        match mesg:
+                            case ("blob:init", meta):
+                                outp.printf(f"Loading blobs file: {meta}")
+                                continue
+                            case ("blob:fini", meta):
+                                outp.printf(f"Finished loading: {meta}")
+                                break
+                            case ("blob", meta):
+                                sha2hex = meta['sha256']
+                                size = meta['size']
+                                sha256 = s_common.uhex(sha2hex)
+                                hasher = hashlib.sha256()
+                                total = 0
+                                bytelist = []
+                                while True:
+                                    try:
+                                        byts = next(msgit)
+                                    except StopIteration:
+                                        mesg = f'Unexpected end of file while reading blob {sha2hex}'
+                                        raise s_exc.BadDataValu(mesg=mesg)
+                                    if type(byts) is tuple:
+                                        msgit = (i for i in [byts] + list(msgit))
+                                        break
+                                    hasher.update(byts)
+                                    total += len(byts)
+                                    bytelist.append(byts)
+                                    if total >= size:
+                                        break
+                                if total != size:
+                                    mesg = f'Blob size mismatch for {sha2hex}: expected {size}, got {total}'
+                                    raise s_exc.BadDataValu(mesg=mesg)
+                                if hasher.digest() != sha256:
+                                    mesg = f'SHA256 mismatch for {sha2hex}'
+                                    raise s_exc.BadDataValu(mesg=mesg)
+                                outp.printf(f'Loading blob {sha2hex} (size={size})')
+                                async with await axon.upload() as upfd:
```

Comment:
We should be able to send the bytes up to an upload object as we stream them out of msgpack; that way the client won't be accumulating them in memory.